src/v/storage/segment.cc

```diff
@@ -202,15 +203,35 @@ ss::future<> segment::release_appender() {
                      a = std::move(a),
                      c = std::move(c),
                      i = std::move(i)]() mutable {
-                        return write_lock().then(
-                          [this,
-                           a = std::move(a),
-                           c = std::move(c),
-                           i = std::move(i)](ss::rwlock::holder h) mutable {
-                              return do_release_appender(
-                                       std::move(a), std::move(c), std::move(i))
-                                .finally([h = std::move(h)] {});
-                          });
+                        return ss::repeat([this,
+                                           a = std::move(a),
+                                           c = std::move(c),
+                                           i = std::move(i)]() mutable {
+                            /**
+                             * we have to use 'spin lock' approach in here singe
+                             * we do not want to make
+                             */
+                            if (_destructive_ops.try_write_lock()) {
```

Comment:
should the `try_write_lock()` return the units too  so you can do `if( auto units = try_lock() )

Reply:
There is a comment on this issue, just few lines above. It is `seastar::semaphore` api. There is no `try_get_units`.